src/Web/Direct/Client.hs

```diff
@@ -0,0 +1,198 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Web.Direct.Client (
+    Client
+  , clientRpcClient
+  , clientPersistedInfo
+  , newClient
+  , setDomains
+  , getDomains
+  , setTalkRooms
+  , getTalkRooms
+  , setMe
+  , getMe
+  , setUsers
+  , getUsers
+  , isActive
+  , inactivate
+  , Channel
+  , withChannel
+  , newChannel
+  , freeChannel
+  , allChannels
+  , findChannel
+  , dispatch
+  , shutdown
+  , sendMessage
+  , send
+  , recv
+  ) where
+
+import qualified Control.Concurrent                       as C
+import qualified Control.Exception                        as E
+import           Control.Monad                            (void)
+import qualified Data.HashMap.Strict                      as HM
+import qualified Data.IORef                               as I
+import qualified Data.MessagePack                         as M
+import qualified Network.MessagePack.RPC.Client.WebSocket as RPC
+
+import           Web.Direct.Message
+import           Web.Direct.PersistedInfo
+import           Web.Direct.Types
+
+----------------------------------------------------------------
+
+data Status = Active | Inactive deriving Eq
+
+type ChannelKey = (TalkId, UserId)
+
+data Channel = Channel (C.MVar (Either Control (Message, Aux))) (C.MVar ()) Client Aux
```

Comment:
Maybe it's time to make a record.
Pattern-matching sometimes prompts us to give inconsistent names to each labels,
while giving record labels recommends us to use the fixed names.
And record labels are better if only one or two fields are accessed (like `dispatch` and `control` do so).

How about these names?

```hs
data Channel = Channel {
      toWorker :: C.MVar (Either Control (Message, Aux))
    , fromWorker :: C.MVar ()
    , channelClient :: Client
    , channelAux :: Aux
}

Reply:
Done.